crates/codegen/grammar/src/parser_definition.rs

```diff
@@ -53,6 +55,25 @@ impl Visitable for TriviaParserDefinitionRef {
     }
 }
 
+/// How many tokens have to be matched to trigger the error recovery.
+/// For ambiguous syntaxes this needs to be set to at least N, where N
+/// is the token lookahead required to disambiguate the syntax.
+///
+// By default, we assume no lookahead is required to recover from
+// unrecognized body between delimiters.
+#[derive(Clone, Debug, Default)]
+pub struct DelimitedRecoveryTokenThreshold(pub u8);
+
+impl From<model::FieldDelimiters> for DelimitedRecoveryTokenThreshold {
+    fn from(delimiters: model::FieldDelimiters) -> Self {
+        Self(
+            delimiters
+                .tokens_matched_acceptance_threshold
+                .unwrap_or(DelimitedRecoveryTokenThreshold::default().0),
```

Comment:
nit suggestion: `#[derive(Default)]` on `DelimitedRecoveryTokenThreshold` will allow `unwrap_or_default()` here as well if you like.